src/timeaxis/timegrid.jl

```diff
@@ -58,21 +58,33 @@ Base.eltype(::Type{<:TimeGrid{T}}) where T = T
 Base.length(tg::TimeGrid{T,P,:finite}) where {T,P} = tg.n
 Base.size(tg::TimeGrid{T,P,:finite}) where{T,P}    = tg.n
 
+###############################################################################
+#  Empty TimeGrid
+###############################################################################
+Base.isempty(tg::TimeGrid{T,P,:finite}) where {T,P} = iszero(tg.n)
+Base.isempty(tg::TimeGrid{T,P,:infinite}) where {T,P} = false
+Base.isfinite(tg::TimeGrid{T,P,:finite}) where {T,P} = true
+Base.isfinite(tg::TimeGrid{T,P,:infinite}) where {T,P} = false
 
 ###############################################################################
 #  Printing
 ###############################################################################
 
-Base.show(io::IO, tg::TimeGrid{T,P,:finite}) where {T,P} =
+function Base.show(io::IO, tg::TimeGrid{T,P,:finite}) where {T,P}
+    isempty(tg) && return
     print(io, "$(tg.o) … $(tg[end]) / $(tg.p) ")
+    return
+end
```

Comment:
hmm, something like this for empty case?
```suggestion
Base.show(io::IO, tg::TimeGrid{T,P,:finite}) where {T,P} = 
    isempty(tg) ? print(io, "(empty)") : print(io, "$(tg.o) … $(tg[end]) / $(tg.p)")
```